lib/vrl/compiler/src/vm/machine.rs

```diff
@@ -0,0 +1,466 @@
+use super::{state::VmState, Variable, VmArgumentList};
+use crate::{vm::argument_list::VmArgument, Context, ExpressionError, Function, Value};
+use diagnostic::Span;
+use std::{collections::BTreeMap, ops::Deref};
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum OpCode {
```

Comment:
I'd like for all of these types to get doc comments explaining what they are, and what they do.